### ISSUE TITLE
[BUG] [ERROR] [django.request] Internal Server Error: /api/search/

### ISSUE THREAD

**Issue by NetRat88** [paperless-ngx/paperless-ngx#6605]
### Description

After updating to the latest version v2.8.1 and when I make a global search nothing happens and the small charging ball then moves continuously:

<img width="1413" alt="screenshot_204" src="https://github.com/paperless-ngx/paperless-ngx/assets/6528507/ac9df9ee-af2f-44c9-9274-81ca20feb6df">

After I checked the paperless-webserver logfiles the following error occurs:

> [2024-05-07 13:14:44,607] [ERROR] [django.request] Internal Server Error: /api/search/
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
>     return self.cursor.execute(sql, params)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/mysql/base.py", line 75, in execute
>     return self.cursor.execute(query, args)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 179, in execute
>     res = self._query(mogrified_query)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 330, in _query
>     db.query(q)
>   File "/usr/local/lib/python3.11/site-packages/MySQLdb/connections.py", line 261, in query
>     _mysql.connection.query(self, query)
> MySQLdb.NotSupportedError: (1235, "This version of MariaDB doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")
> The above exception was the direct cause of the following exception:
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 518, in thread_handler
>     raise exc_info[1]
>   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 42, in inner
>     response = await get_response(request)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 518, in thread_handler
>     raise exc_info[1]
>   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
>     response = await wrapped_callback(
>                ^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 468, in __call__
>     ret = await asyncio.shield(exec_coro)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/asgiref/current_thread_executor.py", line 40, in run
>     result = self.fn(*self.args, **self.kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 522, in thread_handler
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
>     return view_func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
>     return self.dispatch(request, *args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
>     response = self.handle_exception(exc)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
>     self.raise_uncaught_exception(exc)
>   File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
>     raise exc
>   File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
>     response = handler(request, *args, **kwargs)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/paperless/src/documents/views.py", line 1296, in get
>     "total": len(docs)
>              ^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 380, in __len__
>     self._fetch_all()
>   File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 1881, in _fetch_all
>     self._result_cache = list(self._iterable_class(self))
>                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
>     results = compiler.execute_sql(
>               ^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
>     cursor.execute(sql, params)
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
>     return self._execute_with_wrappers(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
>     return executor(sql, params, many, context)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
>     with self.db.wrap_database_errors:
>   File "/usr/local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
>     raise dj_exc_value.with_traceback(traceback) from exc_value
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
>     return self.cursor.execute(sql, params)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/django/db/backends/mysql/base.py", line 75, in execute
>     return self.cursor.execute(query, args)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 179, in execute
>     res = self._query(mogrified_query)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 330, in _query
>     db.query(q)
>   File "/usr/local/lib/python3.11/site-packages/MySQLdb/connections.py", line 261, in query
>     _mysql.connection.query(self, query)
> django.db.utils.NotSupportedError: (1235, "This version of MariaDB doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")

### Steps to reproduce

I used the following Stack:

```
version: "1.0"
services:
  broker:
    image: docker.io/library/redis:7
    restart: always
    volumes:
      - /share/CACHEDEV1_DATA/docker/paperless/redisdata:/data

  db:
    image: docker.io/library/mariadb:10
    restart: always
    volumes:
      - /share/CACHEDEV1_DATA/docker/paperless/dbdata:/var/lib/mysql
    env_file:
      - stack.env

  webserver:
    image: ghcr.io/paperless-ngx/paperless-ngx:latest
    restart: always
    depends_on:
      - db
      - broker
      - gotenberg
      - tika
    ports:
      - "8010:8000"
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - /share/CACHEDEV1_DATA/docker/paperless/data:/usr/src/paperless/data
      - /share/CACHEDEV1_DATA/docker/paperless/media:/usr/src/paperless/media
      - /share/CACHEDEV1_DATA/docker/paperless/export:/usr/src/paperless/export
      - /share/CACHEDEV1_DATA/docker/paperless/scripts:/usr/src/paperless/scripts
      - /share/CACHEDEV2_DATA/Scans/consume:/usr/src/paperless/consume

    env_file:
      - stack.env

  gotenberg:
    image: docker.io/gotenberg/gotenberg:7.8
    restart: always
    # The gotenberg chromium route is used to convert .eml files. We do not
    # want to allow external content like tracking pixels or even javascript.
    command:
      - "gotenberg"
      - "--chromium-disable-javascript=true"
      - "--chromium-allow-list=file:///tmp/.*"

  tika:
    image: ghcr.io/paperless-ngx/tika:latest
    restart: always

```

### Webserver logs

```bash
[2024-05-07 13:14:44,607] [ERROR] [django.request] Internal Server Error: /api/search/
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/mysql/base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 330, in _query
    db.query(q)
  File "/usr/local/lib/python3.11/site-packages/MySQLdb/connections.py", line 261, in query
    _mysql.connection.query(self, query)
MySQLdb.NotSupportedError: (1235, "This version of MariaDB doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = await get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 518, in thread_handler
    raise exc_info[1]
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
    response = await wrapped_callback(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 468, in __call__
    ret = await asyncio.shield(exec_coro)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/asgiref/current_thread_executor.py", line 40, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 522, in thread_handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
    return view_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/views.py", line 1296, in get
    "total": len(docs)
             ^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 380, in __len__
    self._fetch_all()
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 1881, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/usr/local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/backends/mysql/base.py", line 75, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 179, in execute
    res = self._query(mogrified_query)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/MySQLdb/cursors.py", line 330, in _query
    db.query(q)
  File "/usr/local/lib/python3.11/site-packages/MySQLdb/connections.py", line 261, in query
    _mysql.connection.query(self, query)
django.db.utils.NotSupportedError: (1235, "This version of MariaDB doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")
```

### Paperless-ngx version

2.8.1

### Host OS

Linux 5.10.60-qnap

### Installation method

Docker - official image

### Browser

Chrome

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by shamoon** [comment]
Supporting multiple database types is fun.

What version of mariadb are you using?

**Comment by Masgalor** [comment]
The error message here is a little misleading because no version of mysql or mariadb does support this type of query.
[MariaDB subquery limitationss](https://mariadb.com/kb/en/subquery-limitations/)

This is a common problem with django ignoring those limitations.

**Comment by NetRat88** [comment]
But what does it mean for the latest version v2.8.1? It seems that the new global search functions are using subquerys which are not supporting, isn't it? Why else would the search not work properly and the loading bank load endlessly?

I am also using the official supported mariadb version like described in the https://github.com/paperless-ngx/paperless-ngx/blob/dev/docker/compose/docker-compose.mariadb.yml
At least that's what I thought. Or what is currently officially supported? 😅

**Comment by vpmv** [comment]
Sorry to reply in a closed issue, but I just ran into this issue myself. I'm running a similar setup with MariaDB 11. I chose Maria because I use that for all other projects. Spinning up postgres for a single project is a waste of resources.
There's no mention of MariaDB not being supported (for the current release). Is there a release schedule available or is the dev branch already compatible? Not being able to search is quite a hindrance.
